src/traits.jl

```diff
@@ -0,0 +1,31 @@
+# Traits used for statistical models
+
+"""
+    implicit_intercept(T::Type)
+
+Returns `true` if model of type `T` should includes an implicit intercept even
+if none is specified in the formula.  Is `true` by default for all
+`T<:StatisticalModel`, and `false` for others.
```

Comment:
Because the assumption that an intercept column will be included in the model matrix seems pretty specific to statistical modeling, and I want to minimize the number of footguns we provide for people who are using StatsModels in domains where that's not the default assumption.